### PR TITLE
get employee form

### PR DESCRIPTION
**Pull request by Hussein-shahatet97**

### CODE REVIEW

**Comment by Hussein-shahatet97** [comment]
Make a query consist of join between three tables(form,question,employee) using relational tables to get ordered information that show the employee with it's questions,and his answers depend on employee id in any form chosen . 
after define the query to implement this operation, i made the file to connect with database to get a result.

File: server/database/queries/getForms.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	const sql = {
+		text:
+			' SELECT e.name employee_form,f.reviewer_name ,q.context question,eq.result_rate rate_answer,eq.answer context_answer,q.role FROM questions q, employees e,employees_questions eq,form_questions fq,forms f WHERE e.id=eq.employee_id AND q.id=eq.question_id AND fq.form_id=f.id AND fq.question_id=q.id AND eq.employee_id=$1;',
+		values: [employeeID],
+	};
+
+	return db_connection
+		.query(sql)
+		.then((res) => res.rows)
+		.catch((err) => err);
```
Comment (mantagen) [inline server/database/queries/getForms.js]: How will we handle this error?
Has the team discussed a process for handling errors?

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"What does this PR do?" This pull request created to get all form answers when
+the manger select the employee from the list to show result.And this happend by
+employee (id) joined with form (id) ,and questions (id) in the query.
+
+"Does this Fix an Issue?"
+ "Related #34"
+
+"How to test this PR"
+1-You should be able to access database on heroku by url.
+2-When you can access to database by url, implement db_build.sql by db_build.js
```
Comment (yaqootturman) [inline pull_request_template.md]: He should NOT rebuild it, he needs to add the link in config.env file which he should create in root!

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+to create all tables with it's relations. 
+3-Implement dummyData.sql by
+db_build.js. 
+4-Go to check the tables data from terminal by make some basics
+query. 
+5-Make getForm.js consist of function that implement join query between
+form,question,employee in queries file. 
+6-Make route in index.js in controller
+as "(router.get('/api/forminfo/:employeeID', getForms);)".
```
Comment (yaqootturman) [inline pull_request_template.md]: Hussien! He doesn`t need steps you worked on it here, its enough if you say "npm test"

File: server/controller/getForms.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const query = require('../database/queries/getForms');
+exports.getForms = (req, res) => {
+	const employeeID = req.params.employeeID;
+	query(employeeID)
+		.then((data) => {
+			if (data.length != 0) {
+				res.json(data);
+			} else {
+				res.status(404).json({ message: ' Employee form Not Found' });
```
Comment (yaqootturman) [inline server/controller/getForms.js]: Let`s say "The Employee didn`t review/evaluate yet!"

File: server/controller/getForms.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const query = require('../database/queries/getForms');
+exports.getForms = (req, res) => {
+	const employeeID = req.params.employeeID;
```
Comment (yaqootturman) [inline server/controller/getForms.js]: What if you get empty or undefined params? you have to check that before send it to the query!

File: server/database/queries/getForms.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+const getForms = (employeeID) => {
+	const sql = {
+		text:
+			' SELECT e.name employee_form,f.reviewer_name ,q.context question,eq.answer context_answer FROM questions q, employees e,employees_questions eq,form_questions fq,forms f WHERE e.id=eq.employee_id AND q.id=eq.question_id AND fq.form_id=f.id AND fq.question_id=q.id AND eq.employee_id=$1;',
+		values: [employeeID],
+	};
+
+	return db_connection
+		.query(sql)
+		.then((res) => res.rows)
```
Comment (yaqootturman) [inline server/database/queries/getForms.js]: We talked about the promise in the query, if you don't have a clear idea yes, let us discuss again, please!

File: server/database/queries/getForms.js (hunk 1 above)
Comment (yaqootturman) [reply server/database/queries/getForms.js]: We will handle errors by middlewares, we will accept consol.logs now and the next step will be the middleware!

File: server/database/queries/getForms.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const db_connection = require('../config/db_connection');
+
+const getForms = (employeeID) => {
+	const sql = {
+		text:
+			' SELECT e.name employee_form,f.reviewer_name ,q.context question,eq.answer context_answer FROM questions q, employees e,employees_questions eq,form_questions fq,forms f WHERE e.id=eq.employee_id AND q.id=eq.question_id AND fq.form_id=f.id AND fq.question_id=q.id AND eq.employee_id=$1;',
```
Comment (yaqootturman) [inline server/database/queries/getForms.js]: Great!

**Review by yaqootturman — CHANGES_REQUESTED** [review]

File: pull_request_template.md (hunk 2 above)
Comment (Hussein-shahatet97) [reply pull_request_template.md]: I made brief description without rebuild .

File: pull_request_template.md (hunk 3 above)
Comment (Hussein-shahatet97) [reply pull_request_template.md]: done!

File: server/controller/getForms.js (hunk 4 above)
Comment (Hussein-shahatet97) [reply server/controller/getForms.js]: done !

File: server/controller/getForms.js (hunk 5 above)
Comment (Hussein-shahatet97) [reply server/controller/getForms.js]: done !

File: server/database/queries/getForms.js (hunk 6 above)
Comment (Hussein-shahatet97) [reply server/database/queries/getForms.js]: done !

**Review by yaqootturman — DISMISSED** [review]

**Review by rubasider — DISMISSED** [review]

File: server/controller/forms/getForms.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+			.then((data) => {
+				if (data.length != 0) {
+					res.json(data);
+				} else {
+					res
+						.status(404)
+						.json({ message: ' Employee did not  evaluated yet!' });
+				}
+			})
+			.catch((err) => err);
```
Comment (mantagen) [inline server/controller/forms/getForms.js]: Please handle this error ... something like
```
.catch(err => res.status(500).json({ message: "An unknown error occurred" }))
```

**Review by mantagen — DISMISSED** [review]
Approve.
BUT FIRST PLEASE HANDLE THE ERROR

**Review by rubasider — APPROVED** [review]

**Review by yaqootturman — APPROVED** [review]

**Pull request merged**